robot/web/web.go

```diff
@@ -365,42 +364,43 @@ func (svc *webService) updateResources(resources map[resource.Name]resource.Reso
 		groupedResources[n.API] = r
 	}
 
-	apiRegs := resource.RegisteredAPIs()
-	for a, v := range groupedResources {
-		apiResColl, ok := svc.services[a]
-		// TODO(RSDK-144): register new service if it doesn't currently exist
+	// All known APIs have pre-registered resource collections, so loop through them. We want to empty out any collections if
+	// there are no longer resources available in that API.
+	for api, coll := range svc.services {
```

Comment:
the logic here was messy because we allowed the possibility of having a group of resources that do not have a pre-existing resource collection. By removing that path (which was never used anyway), the logic can be generally simplified

Reply:
I think I follow? Does this mean something like: a robot config hypothetically having a "fake motor" model that it's version of the rdk/viam-server is happy to create and configure. But there was no corresponding `motor/server.go` file that registered the "motor API" and create its corresponding "[motor resource collection](https://github.com/viamrobotics/rdk/blob/ea997e052cbfb9e9a72984696347062bb1451abb/components/motor/server.go#L17)"?

Reply:
yep - the case is usually for a custom api, and we're adding it to a collection that never gets used anyway